pkg/controller/bastion/options.go

```diff
@@ -163,30 +163,98 @@ func determineImageID(shoot *gardencorev1beta1.Shoot, providerConfig *awsv1alpha
 	return "", fmt.Errorf("found no suitable AMI for machines in region %q", shoot.Spec.Region)
 }
 
-func determineInstanceType(ctx context.Context, awsClient *awsclient.Client) (string, error) {
-	offerings, err := awsClient.EC2.DescribeInstanceTypeOfferingsWithContext(ctx, &ec2.DescribeInstanceTypeOfferingsInput{})
+func determineInstanceType(ctx context.Context, imageID string, awsClient *awsclient.Client) (string, error) {
+	var preferredType string
+	imageInfo, err := getImages(ctx, imageID, awsClient)
 	if err != nil {
-		return "", fmt.Errorf("failed to list instance types: %w", err)
+		return "", err
 	}
 
-	types := make([]string, len(offerings.InstanceTypeOfferings))
-	for i, offering := range offerings.InstanceTypeOfferings {
-		types[i] = *offering.InstanceType
+	if imageInfo.Architecture == nil {
+		return "", fmt.Errorf("image architecture is empty")
 	}
 
-	// prefer t2.nano
-	for _, t := range types {
-		if t == "t2.nano" {
-			return t, nil
-		}
+	imageArchitecture := imageInfo.Architecture
+
+	// default instance type
+	switch *imageArchitecture {
+	case "x86_64":
+		preferredType = "t2.nano"
+	case "arm64":
+		preferredType = "t4g.nano"
+	default:
+		return "", fmt.Errorf("image architecture not supported")
 	}
 
-	// fallback to the first (hopefully smallest) other general purpose instance type
-	for _, t := range types {
-		if strings.HasPrefix(t, "t") {
-			return t, nil
-		}
+	exist, err := getInstanceTypeOfferings(ctx, preferredType, awsClient)
+	if err != nil {
+		return "", err
+	}
+
+	if len(exist.InstanceTypeOfferings) != 0 {
+		return preferredType, nil
+	}
+
+	// filter t type instance
+	tTypes, err := getInstanceTypeOfferings(ctx, "t*", awsClient)
+	if err != nil {
+		return "", err
+	}
+
+	if len(tTypes.InstanceTypeOfferings) == 0 {
+		return "", fmt.Errorf("no t* instance type offerings available")
+	}
+
+	tTypeSet := sets.NewString()
+	for _, t := range tTypes.InstanceTypeOfferings {
+		tTypeSet.Insert(*t.InstanceType)
+	}
+
+	result, err := awsClient.EC2.DescribeInstanceTypes(&ec2.DescribeInstanceTypesInput{
+		InstanceTypes: aws.StringSlice(tTypeSet.UnsortedList()),
+		Filters: []*ec2.Filter{
+			{
+				Name:   aws.String("processor-info.supported-architecture"),
+				Values: []*string{imageArchitecture},
+			},
+		},
+	})
+
+	if err != nil {
+		return "", err
+	}
+
+	if len(result.InstanceTypes) == 0 {
+		return "", fmt.Errorf("no instance types returned for architecture %s and instance types list %v", *imageArchitecture, tTypeSet.UnsortedList())
+	}
+
+	return *result.InstanceTypes[0].InstanceType, nil
```

Comment:
same here, is `InstanceType` guaranteed to be not nil? otherwise this PR looks good to me

Reply:
b93e187bfa57280390f5dcbc9f2fc1d9522b5529